src/Utils/TreeUtils.test.ts

```diff
@@ -0,0 +1,32 @@
+
+        test('inTree', () => {
+            const logRanker = new TreeUtils([trainDialog1, trainDialog2, trainDialog3], [])
```

Comment:
It's a weird error that this is named `logRanker`. I assume it was intentional since I can't imagine how you would not notice unless really copy paste, but it's very weird to me to construct instance of `TreeUtils` and somehow get an instance of `logRanker`.

I would expect `const treeUtil = new TreeUtil()`

Then the methods `isInTree` would make more sense on a tree instance than a log ranker instance.

Reply:
nice catch.  missed this on refactor